File: android/app/src/main/java/com/nhs/online/nhsonline/support/ApplicationState.kt
package com.nhs.online.nhsonline.support

import org.threeten.bp.LocalDateTime

class ApplicationState(private val menuTimeout: Long) {

    private var lastInteraction = LocalDateTime.MIN

    fun block() {
        lastInteraction = LocalDateTime.now()
    }

    fun unBlock() {
        lastInteraction = LocalDateTime.MIN
    }

    fun isReady(): Boolean {
        return LocalDateTime.now().minusSeconds(menuTimeout) > lastInteraction
    }
}

File: ios/NHSOnline/NHSOnline/Application/Lifecycle/ApplicationState.swift
import Foundation

class ApplicationState {
    var lastInteraction = Date.distantPast
    
    public func block() {
        lastInteraction = Date()
    }
    
    public func unBlock() {
        lastInteraction = Date.distantPast
    }
    
    public func isReady() -> Bool {
        return Date().timeIntervalSince(lastInteraction) > config().MenuTimeoutSeconds
    }
}
